language/functional_tests/src/utils.rs

```diff
@@ -22,20 +22,24 @@ pub fn substitute_addresses(config: &GlobalConfig, text: &str) -> String {
     PAT.replace_all(text, |caps: &Captures| {
         let name = &caps[1];
 
-        if let Some(account) = config.get_account_for_name(name) {
-            format!("0x{}", account.address())
-        } else {
-            panic!(
-                "account '{}' does not exist, cannot substitute address",
-                name
-            )
-        }
+        format!("0x{}", config.get_account_for_name(name).unwrap().address())
```

Comment:
What are your thoughts on using e.g. [`UpperHex`](https://doc.rust-lang.org/std/fmt/trait.UpperHex.html) (or `LowerHex`) here instead of `"0x{}"`?

Reply:
Tested, didn't work. In fact, the implementations in `types::account_address::AccountAddress` are wrong. 
```
impl fmt::Display for AccountAddress {
    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
        // Forward to the LowerHex impl with a "0x" prepended (the # flag).
        write!(f, "{:#x}", self)
    }
}

impl fmt::Debug for AccountAddress {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        // Forward to the LowerHex impl with a "0x" prepended (the # flag).
        write!(f, "{:#x}", self)
    }
}

impl fmt::LowerHex for AccountAddress {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{}", hex::encode(&self.0))
    }
}
```
Since a manual implementation of `LowerHex` is provided, the `#` modifier no longer has any meaning. (I don't know if there's a way to detect the presence of `#` when implementing `LowerHex`.) All three functions prints out the hex value without "0x".

Let's fix this in a separate diff.